src/data/perf_stat.rs

```diff
@@ -127,55 +156,53 @@ impl CollectData for PerfStatRaw {
 
         cfg_if::cfg_if! {
             if #[cfg(target_arch = "aarch64")] {
-                let perf_list = grv_perf_events::PERF_LIST.to_vec();
+                let mut perf_list = to_events(arm64_perf_list::GRV_EVENTS)?;
             } else if #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] {
-                let platform_specific_counter;
-                let cpu_info = get_cpu_info()?;
-                let mut perf_list;
+                let cpu_info = crate::data::utils::get_cpu_info()?;
+                let platform_specific_counter: &[u8];
+                let base: &[u8];
 
                 /* Get Vendor Specific Perf events */
                 if cpu_info.vendor == "GenuineIntel" {
-                    perf_list = intel_perf_events::PERF_LIST.to_vec();
+                    base = x86_perf_list::INTEL_EVENTS;
 
                     /* Get Model specific events */
                     platform_specific_counter = match cpu_info.model_name.as_str() {
-                        "Intel(R) Xeon(R) Platinum 8375C CPU @ 2.90GHz" => ICX_CTRS.to_vec(),
-                        "Intel(R) Xeon(R) Platinum 8488C" => SPR_CTRS.to_vec(),
-                        _ => Vec::new(),
+                        "Intel(R) Xeon(R) Platinum 8375C CPU @ 2.90GHz" => x86_perf_list::ICX_CTRS,
```

Comment:
Unrelated to this change I realize, but the @ 2.90GHz seems needlessly precise.

Reply:
These are based on https://github.com/aws/aws-graviton-getting-started/blob/main/perfrunbook/utilities/measure_and_plot_basic_pmu_counters.py#L183